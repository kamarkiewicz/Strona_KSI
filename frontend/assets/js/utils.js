
// Access image field from backend
function getImage (image) {
  if ((image || {}).path) {
    // path may be relative URL
    let isAbsolute = /^(?:[a-z]+:)?\/\//i.test(image.path)
    image.path = (isAbsolute ? '' : process.env.API_URL) + image.path
  }
  return image
}

export { getImage };
