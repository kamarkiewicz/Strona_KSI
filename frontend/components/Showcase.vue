<template>
  <section class="showcase bg-white">
    <div class="container pt-slant pb-slant">

      <div class="row pt-4" v-for="(el, index) in entries" :key="index">
        <div class="col-md-5 align-self-center"
            :class="{ 'order-md-2': index % 2 === 0 }">
          <img class="img-fluid d-block mb-4 w-100 img-thumbnail" :src="el.image.src" :alt="el.image.title">
        </div>
        <div class="col-md-7">
          <h2 class="text-primary pt-3" v-text="el.title"></h2>
          <p class="description" v-text="el.description"></p>
          <nuxt-link class="pretty-link" :to="localePath(el.link)" v-text="$t('general.readMore')"></nuxt-link>
        </div>
      </div>

    </div>
  </section>
</template>

<script>
import { Image } from '~/assets/js/utils'

export default {
  props: {
    entries: {
      type: Array,
      required: true
    },
  },
}
</script>
