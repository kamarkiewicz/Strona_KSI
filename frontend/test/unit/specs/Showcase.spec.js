import Vue from 'vue'
import { mount } from '@vue/test-utils'
import Showcase from '~/components/Showcase.vue'
import { Image } from '~/assets/js/utils'

Vue.config.ignoredElements = [
  'b-row', 'b-col', 'nuxt-link'
];

describe('Showcase', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Showcase, {
      mocks: {
        $t: (label) => { return label },
        $d: (label) => { return label },
        localePath: (args) => { return JSON.stringify(args) },
      },
      propsData: {
        entries: [
          {
            id: 1,
            title: 'O nas. Historia KSI',
            description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
            image: new Image('https://picsum.photos/270/180?image=985'),
            link: 'about-us',
          },
          {
            id: 2,
            title: 'Projekty',
            description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
            image: new Image('https://picsum.photos/270/180?image=986'),
            link: 'projects',
          },
          {
            id: 3,
            title: 'CzÅ‚onkowie',
            description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
            image: new Image('https://picsum.photos/270/180?image=987'),
            link: 'members',
          },
        ],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
