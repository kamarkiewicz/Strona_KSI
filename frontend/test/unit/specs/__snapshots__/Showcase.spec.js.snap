// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase renders correctly 1`] = `
<section class="showcase bg-white">
  <div class="container">
    <div class="row pt-4">
      <div class="col-md-5 align-self-center order-md-2">
        <img src="https://picsum.photos/270/180?image=985" alt="Featured image" class="img-fluid d-block mb-4 w-100 img-thumbnail">
      </div>
      <div class="col-md-7">
        <h2 class="text-primary pt-3">O nas. Historia KSI</h2>
        <p class="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
          dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      </div>
    </div>
    <div class="row pt-4">
      <div class="col-md-5 align-self-center">
        <img src="https://picsum.photos/270/180?image=986" alt="Featured image" class="img-fluid d-block mb-4 w-100 img-thumbnail">
      </div>
      <div class="col-md-7">
        <h2 class="text-primary pt-3">Projekty</h2>
        <p class="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
          dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      </div>
    </div>
    <div class="row pt-4">
      <div class="col-md-5 align-self-center order-md-2">
        <img src="https://picsum.photos/270/180?image=987" alt="Featured image" class="img-fluid d-block mb-4 w-100 img-thumbnail">
      </div>
      <div class="col-md-7">
        <h2 class="text-primary pt-3">Cz≈Çonkowie</h2>
        <p class="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
          dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      </div>
    </div>
  </div>
</section>
`;
